pkg/aspect/applicationLogsManager.go

```diff
@@ -108,107 +128,100 @@ func (applicationLogsManager) ValidateConfig(c adapter.AspectConfig) (ce *adapte
 
 func (e *applicationLogsWrapper) Execute(attrs attribute.Bag, mapper expr.Evaluator, ma APIMethodArgs) Output {
+	result := &multierror.Error{}
 	var entries []adapter.LogEntry
 
-	// TODO: would be nice if we could use a mutable.Bag here and could pass it around
-	// labels holds the generated attributes from mapper
-	labels := make(map[string]interface{})
-	for attr, exp := range e.inputs {
-		if val, err := mapper.Eval(exp, attrs); err == nil {
-			labels[attr] = val
+	for name, md := range e.metadata {
+		labels, err := evalAll(md.labels, attrs, mapper)
+		if err != nil {
+			result = multierror.Append(result, fmt.Errorf("failed to eval labels for log entry '%s' with err: %s", name, err))
+			continue
 		}
-	}
 
-	for _, d := range e.descriptors {
-		entry := adapter.LogEntry{
-			LogName:   e.logName,
-			Labels:    make(map[string]interface{}),
-			Severity:  severityVal(e.severityAttribute, attrs, labels),
-			Timestamp: timeVal(e.timestampAttribute, attrs, labels, e.defaultTimeFn()).Format(e.timestampFmt),
+		templateVals, err := evalAll(md.tmplExprs, attrs, mapper)
+		if err != nil {
+			result = multierror.Append(result, fmt.Errorf("failed to eval template values for log entry '%s' with err: %s", name, err))
+			continue
 		}
 
-		payloadStr := stringVal(d.PayloadAttribute, attrs, labels, "")
-		switch d.PayloadFormat {
-		case dpb.TEXT:
-			entry.TextPayload = payloadStr
-		case dpb.JSON:
-			err := json.Unmarshal([]byte(payloadStr), &entry.StructPayload)
-			if err != nil {
-				return Output{Status: status.WithError(fmt.Errorf("could not unmarshal json payload: %v", err))}
-			}
+		buf := pool.GetBuffer()
+		err = md.tmpl.Execute(buf, templateVals)
+		if err != nil {
+			pool.PutBuffer(buf)
+			result = multierror.Append(result, fmt.Errorf(
+				"failed to construct payload string for log entry '%s' with template execution err: %s", name, err))
+			continue
 		}
 
-		for _, a := range d.Attributes {
-			if a == d.PayloadAttribute {
-				continue
-			}
-			if val, ok := labels[a]; ok {
-				entry.Labels[a] = val
+		sevStr, err := mapper.EvalString(md.severity, attrs)
+		if err != nil {
+			result = multierror.Append(result,
+				fmt.Errorf("failed to eval severity for log entry '%s', continuing with DEFAULT severity. Eval err: %s", name, err))
+			sevStr = ""
+		}
+		// If we can't parse the string we'll get the default severity, so we don't care about the success return.
+		severity, _ := adapter.SeverityByName(sevStr)
+
+		et, err := mapper.Eval(md.timestamp, attrs)
+		if err != nil {
+			result = multierror.Append(result,
+				fmt.Errorf("failed to eval time for log entry '%s', continuing with time.Now(). Eval err: %s", name, err))
+			et = time.Now()
+		}
+		t, _ := et.(time.Time) // we don't check the cast because expression type checking ensures we get a time.
+
+		entry := adapter.LogEntry{
+			LogName:   e.name,
+			Labels:    labels,
+			Timestamp: t.String(),
```

Comment:
i asked this above, but this feels like we've lost the ability to specify how the timestamp is formatted in the logs. Is that the case?